torch/nn/modules/batchnorm.py

```diff
@@ -440,19 +437,24 @@ def forward(self, input):
             else:  # use exponential moving average
                 exponential_average_factor = self.momentum
 
-        world_size = 1
-        process_group = torch.distributed.group.WORLD
-        if self.process_group:
-            process_group = self.process_group
-        world_size = torch.distributed.get_world_size(process_group)
+        need_sync = self.training or not self.track_running_stats
+        if need_sync:
+            process_group = torch.distributed.group.WORLD
+            if self.process_group:
+                process_group = self.process_group
+            world_size = torch.distributed.get_world_size(process_group)
```

Comment:
[`process_group.size()`](https://github.com/pytorch/pytorch/blob/master/torch/csrc/distributed/c10d/init.cpp#L220) should work as well.

Reply:
From the implementation of `get_group_size` it looks like `.size()` is not sufficient?

https://github.com/pytorch/pytorch/blob/dfdb86a59577d8b0fc4565988a8ac01b5ecd339f/torch/distributed/distributed_c10d.py#L194-L204